proposals/Exceptions.md

```diff
@@ -170,21 +174,104 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction always re-throws the exception caught by the referenced
+catch block. This allows the catch block to clean up state before the exception
+is passed back to the next enclosing try block.
+
+Associated with the _rethrow_ instruction is a _label_. The label is used to
+disambiguate which exception is to be rethrown, when inside nested catch blocks.
+
+The label is the relative block depth to the corresponding try block for which
+the catching block appears.
+
+For example consider the following example:
+
+```
+try
+  ...
+catch 1
+  ...
+  begin
+    ...
+    try
+      ...
+    catch 2
+      ...
+      try
+        ...
+      catch 3
+        ...
+        rethrow N
+      end
+    end
+  end
+  ...
+end
+```
+
+In this example, `N` is used to disambiguate which caught exception is being
+rethrown. It could rethrow any of the three caught expceptions. Hence,
+`rethrow 0` corresponds to the exception caught by `catch 3`, `rethrow 1`
+corresponds to the exception caught by `catch 2`, and `rethrow 4` corresponds
+to the exception caught by `catch 1`.
 
 ### Debugging
 
 Earlier discussion implied that when an exception is thrown, the runtime will
-pop the value stack across function calls until a corresponding, enclosing try
+pop the operand stack across function calls until a corresponding, enclosing try
 block is found. The implementation may actually not do this. Rather, it may
 first search up the call stack to see if there is an enclosing try. If none are
-found, it could terminate the thread at that point of the throw. This would
+found, it could terminate the thread at the point of the throw. This would
 allow better debugging capability, since the corresponding call stack is still
 there to query.
 
+## Changes to the text format.
+
+This section describes change in the
+[instruction syntax document](https://github.com/WebAssembly/spec/blob/master/document/text/instructions.rst).
+
+### Control Instructions
+
+The following rule is added to *instructions*:
+
+```
+instructions ::=
+  ...
+  try resulttype instr* catch+ end |
+  throw except_index |
+  rethrow catch_index
+  
+catch ::=
+  catch except_index inst* |
+  catch_all inst*
+```
+
+Like the *block*, *loop*, and *if* instructions, the *try* instruction is a
+*structured* instruction, and is implicitly labeled. this allows branch
+instructions to exit try blocks.
+
+The `except_index` of the `catch` instruction is the exception tag for the caught
+exception. Similarly, the `except_index` of the *throw* instruction is the tag for
```

Comment:
Instructions are sometimes in italics (*throw*), sometimes set as code (`throw`). You might want to harmonize that.